pkg/dsdk/user_data.go

```diff
@@ -0,0 +1,100 @@
+package dsdk
+
+import (
+	"context"
+	greq "github.com/levigross/grequests"
+	_path "path"
+)
+
+type UserData struct {
+	AppInstanceId  string                  `json:"app_instance_id"`
+	Data           map[string]interface{}  `json:"data"`
+}
+
+type UserDatas struct {
+	Path string
+}
+
+func newUserDatas(path string) *UserDatas {
+	return &UserDatas{
+		Path: _path.Join(path, "user_data"),
+	}
+}
+
+type UserDataUpdateRequest struct {
+	Ctxt                 context.Context        `json:"-"`
+	AppInstanceId        string                 `json:"app_instance_id" mapstructure:"app_instance_id"`
+	Data                 map[string]interface{} `json:"data" mapstructure:"data"`
+}
+
+// Update adds a JSON User Data Record to an App Instance
+func (e *UserDatas) Update(ud *UserDataUpdateRequest) (*UserData, *ApiErrorResponse, error) {
```

Comment:
Maybe rename `Update` and the associated request TDO to use `Set` instead to fit with the rest of the SDK?